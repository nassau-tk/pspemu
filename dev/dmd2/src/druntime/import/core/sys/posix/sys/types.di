// D import file generated from 'src\core\sys\posix\sys\types.d'
module core.sys.posix.sys.types;
private import core.sys.posix.config;

private import core.stdc.stdint;

public import core.stdc.stddef;

public import core.stdc.time;

extern (C) 
{
    version (linux)
{
    static if(__USE_FILE_OFFSET64)
{
    alias long blkcnt_t;
    alias ulong ino_t;
    alias long off_t;
}
else
{
    alias c_long blkcnt_t;
    alias c_ulong ino_t;
    alias c_long off_t;
}
    alias c_long blksize_t;
    alias ulong dev_t;
    alias uint gid_t;
    alias uint mode_t;
    alias c_ulong nlink_t;
    alias int pid_t;
    alias c_long ssize_t;
    alias uint uid_t;
}
else
{
    version (OSX)
{
    alias long blkcnt_t;
    alias int blksize_t;
    alias int dev_t;
    alias uint gid_t;
    alias uint ino_t;
    alias ushort mode_t;
    alias ushort nlink_t;
    alias long off_t;
    alias int pid_t;
    alias c_long ssize_t;
    alias uint uid_t;
}
else
{
    version (FreeBSD)
{
    alias long blkcnt_t;
    alias uint blksize_t;
    alias uint dev_t;
    alias uint gid_t;
    alias uint ino_t;
    alias ushort mode_t;
    alias ushort nlink_t;
    alias long off_t;
    alias int pid_t;
    alias c_long ssize_t;
    alias uint uid_t;
    alias uint fflags_t;
}
}
}
    version (linux)
{
    static if(__USE_FILE_OFFSET64)
{
    alias ulong fsblkcnt_t;
    alias ulong fsfilcnt_t;
}
else
{
    alias c_ulong fsblkcnt_t;
    alias c_ulong fsfilcnt_t;
}
    alias uint id_t;
    alias int key_t;
    alias c_long suseconds_t;
    alias uint useconds_t;
}
else
{
    version (OSX)
{
    alias uint fsblkcnt_t;
    alias uint fsfilcnt_t;
    alias uint id_t;
    alias int suseconds_t;
    alias uint useconds_t;
}
else
{
    version (FreeBSD)
{
    alias ulong fsblkcnt_t;
    alias ulong fsfilcnt_t;
    alias long id_t;
    alias c_long key_t;
    alias c_long suseconds_t;
    alias uint useconds_t;
}
}
}
    version (linux)
{
    version (X86)
{
    enum __SIZEOF_PTHREAD_ATTR_T = 36;
    enum __SIZEOF_PTHREAD_MUTEX_T = 24;
    enum __SIZEOF_PTHREAD_MUTEXATTR_T = 4;
    enum __SIZEOF_PTHREAD_COND_T = 48;
    enum __SIZEOF_PTHREAD_CONDATTR_T = 4;
    enum __SIZEOF_PTHREAD_RWLOCK_T = 32;
    enum __SIZEOF_PTHREAD_RWLOCKATTR_T = 8;
    enum __SIZEOF_PTHREAD_BARRIER_T = 20;
    enum __SIZEOF_PTHREAD_BARRIERATTR_T = 4;
}
else
{
    version (X86_64)
{
    enum __SIZEOF_PTHREAD_ATTR_T = 56;
    enum __SIZEOF_PTHREAD_MUTEX_T = 40;
    enum __SIZEOF_PTHREAD_MUTEXATTR_T = 4;
    enum __SIZEOF_PTHREAD_COND_T = 48;
    enum __SIZEOF_PTHREAD_CONDATTR_T = 4;
    enum __SIZEOF_PTHREAD_RWLOCK_T = 56;
    enum __SIZEOF_PTHREAD_RWLOCKATTR_T = 8;
    enum __SIZEOF_PTHREAD_BARRIER_T = 32;
    enum __SIZEOF_PTHREAD_BARRIERATTR_T = 4;
}
}
    union pthread_attr_t
{
    byte[__SIZEOF_PTHREAD_ATTR_T] __size;
    c_long __align;
}
    private alias int __atomic_lock_t;

    private struct _pthread_fastlock
{
    c_long __status;
    __atomic_lock_t __spinlock;
}

    private alias void* _pthread_descr;

    union pthread_cond_t
{
    byte[__SIZEOF_PTHREAD_COND_T] __size;
    long __align;
}
    union pthread_condattr_t
{
    byte[__SIZEOF_PTHREAD_CONDATTR_T] __size;
    int __align;
}
    alias uint pthread_key_t;
    union pthread_mutex_t
{
    byte[__SIZEOF_PTHREAD_MUTEX_T] __size;
    c_long __align;
}
    union pthread_mutexattr_t
{
    byte[__SIZEOF_PTHREAD_MUTEXATTR_T] __size;
    int __align;
}
    alias int pthread_once_t;
    struct pthread_rwlock_t
{
    byte[__SIZEOF_PTHREAD_RWLOCK_T] __size;
    c_long __align;
}
    struct pthread_rwlockattr_t
{
    byte[__SIZEOF_PTHREAD_RWLOCKATTR_T] __size;
    c_long __align;
}
    alias c_ulong pthread_t;
}
else
{
    version (OSX)
{
    version (X86_64)
{
    enum __PTHREAD_SIZE__ = 1168;
    enum __PTHREAD_ATTR_SIZE__ = 56;
    enum __PTHREAD_MUTEXATTR_SIZE__ = 8;
    enum __PTHREAD_MUTEX_SIZE__ = 56;
    enum __PTHREAD_CONDATTR_SIZE__ = 8;
    enum __PTHREAD_COND_SIZE__ = 40;
    enum __PTHREAD_ONCE_SIZE__ = 8;
    enum __PTHREAD_RWLOCK_SIZE__ = 192;
    enum __PTHREAD_RWLOCKATTR_SIZE__ = 16;
}
else
{
    version (X86)
{
    enum __PTHREAD_SIZE__ = 596;
    enum __PTHREAD_ATTR_SIZE__ = 36;
    enum __PTHREAD_MUTEXATTR_SIZE__ = 8;
    enum __PTHREAD_MUTEX_SIZE__ = 40;
    enum __PTHREAD_CONDATTR_SIZE__ = 4;
    enum __PTHREAD_COND_SIZE__ = 24;
    enum __PTHREAD_ONCE_SIZE__ = 4;
    enum __PTHREAD_RWLOCK_SIZE__ = 124;
    enum __PTHREAD_RWLOCKATTR_SIZE__ = 12;
}
}
    struct pthread_handler_rec
{
    void function(void*) __routine;
    void* __arg;
    pthread_handler_rec* __next;
}
    struct pthread_attr_t
{
    c_long __sig;
    byte[__PTHREAD_ATTR_SIZE__] __opaque;
}
    struct pthread_cond_t
{
    c_long __sig;
    byte[__PTHREAD_COND_SIZE__] __opaque;
}
    struct pthread_condattr_t
{
    c_long __sig;
    byte[__PTHREAD_CONDATTR_SIZE__] __opaque;
}
    alias c_ulong pthread_key_t;
    struct pthread_mutex_t
{
    c_long __sig;
    byte[__PTHREAD_MUTEX_SIZE__] __opaque;
}
    struct pthread_mutexattr_t
{
    c_long __sig;
    byte[__PTHREAD_MUTEXATTR_SIZE__] __opaque;
}
    struct pthread_once_t
{
    c_long __sig;
    byte[__PTHREAD_ONCE_SIZE__] __opaque;
}
    struct pthread_rwlock_t
{
    c_long __sig;
    byte[__PTHREAD_RWLOCK_SIZE__] __opaque;
}
    struct pthread_rwlockattr_t
{
    c_long __sig;
    byte[__PTHREAD_RWLOCKATTR_SIZE__] __opaque;
}
    private struct _opaque_pthread_t
{
    c_long __sig;
    pthread_handler_rec* __cleanup_stack;
    byte[__PTHREAD_SIZE__] __opaque;
}

    alias _opaque_pthread_t* pthread_t;
}
else
{
    version (FreeBSD)
{
    alias int lwpid_t;
    alias void* pthread_attr_t;
    alias void* pthread_cond_t;
    alias void* pthread_condattr_t;
    alias void* pthread_key_t;
    alias void* pthread_mutex_t;
    alias void* pthread_mutexattr_t;
    alias void* pthread_once_t;
    alias void* pthread_rwlock_t;
    alias void* pthread_rwlockattr_t;
    alias void* pthread_t;
}
}
}
    version (linux)
{
    struct pthread_barrier_t
{
    byte[__SIZEOF_PTHREAD_BARRIER_T] __size;
    c_long __align;
}
    struct pthread_barrierattr_t
{
    byte[__SIZEOF_PTHREAD_BARRIERATTR_T] __size;
    int __align;
}
}
else
{
    version (FreeBSD)
{
    alias void* pthread_barrier_t;
    alias void* pthread_barrierattr_t;
}
}
    version (linux)
{
    alias int pthread_spinlock_t;
}
else
{
    version (OSX)
{
}
else
{
    version (FreeBSD)
{
    alias void* pthread_spinlock_t;
}
}
}
}
