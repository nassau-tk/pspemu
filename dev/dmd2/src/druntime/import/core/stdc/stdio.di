// D import file generated from 'src\core\stdc\stdio.d'
module core.stdc.stdio;
private 
{
    import core.stdc.config;
    import core.stdc.stddef;
    import core.stdc.stdarg;
    version (FreeBSD)
{
    import core.sys.posix.sys.types;
}
}
extern (C) nothrow 
{
    version (Windows)
{
    enum 
{
BUFSIZ = 16384,
EOF = -1,
FOPEN_MAX = 20,
FILENAME_MAX = 256,
TMP_MAX = 32767,
SYS_OPEN = 20,
}
    enum int _NFILE = 60;
    enum string _P_tmpdir = "\\";
    enum wstring _wP_tmpdir = "\\";
    enum int L_tmpnam = _P_tmpdir.length + 12;
}
else
{
    version (linux)
{
    enum 
{
BUFSIZ = 8192,
EOF = -1,
FOPEN_MAX = 16,
FILENAME_MAX = 4095,
TMP_MAX = 238328,
L_tmpnam = 20,
}
}
else
{
    version (OSX)
{
    enum 
{
BUFSIZ = 1024,
EOF = -1,
FOPEN_MAX = 20,
FILENAME_MAX = 1024,
TMP_MAX = 308915776,
L_tmpnam = 1024,
}
    private 
{
    struct __sbuf
{
    ubyte* _base;
    int _size;
}
    struct __sFILEX
{
}
}
}
else
{
    version (FreeBSD)
{
    enum 
{
EOF = -1,
FOPEN_MAX = 20,
FILENAME_MAX = 1024,
TMP_MAX = 308915776,
L_tmpnam = 1024,
}
    struct __sbuf
{
    ubyte* _base;
    int _size;
}
    alias _iobuf __sFILE;
    union __mbstate_t
{
    char[128] _mbstate8;
    long _mbstateL;
}
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}
    enum 
{
SEEK_SET,
SEEK_CUR,
SEEK_END,
}
    struct _iobuf
{
    align (1)version (Windows)
{
    char* _ptr;
    int _cnt;
    char* _base;
    int _flag;
    int _file;
    int _charbuf;
    int _bufsiz;
    int __tmpnum;
}
else
{
    version (linux)
{
    int _flags;
    char* _read_ptr;
    char* _read_end;
    char* _read_base;
    char* _write_base;
    char* _write_ptr;
    char* _write_end;
    char* _buf_base;
    char* _buf_end;
    char* _save_base;
    char* _backup_base;
    char* _save_end;
    void* _markers;
    _iobuf* _chain;
    int _fileno;
    int _blksize;
    int _old_offset;
    ushort _cur_column;
    byte _vtable_offset;
    char[1] _shortbuf;
    void* _lock;
}
else
{
    version (OSX)
{
    ubyte* _p;
    int _r;
    int _w;
    short _flags;
    short _file;
    __sbuf _bf;
    int _lbfsize;
    int* function(void*) _close;
    int* function(void*, char*, int) _read;
    fpos_t* function(void*, fpos_t, int) _seek;
    int* function(void*, char*, int) _write;
    __sbuf _ub;
    __sFILEX* _extra;
    int _ur;
    ubyte[3] _ubuf;
    ubyte[1] _nbuf;
    __sbuf _lb;
    int _blksize;
    fpos_t _offset;
}
else
{
    version (FreeBSD)
{
    ubyte* _p;
    int _r;
    int _w;
    short _flags;
    short _file;
    __sbuf _bf;
    int _lbfsize;
    void* _cookie;
    int function(void*) _close;
    int function(void*, char*, int) _read;
    fpos_t function(void*, fpos_t, int) _seek;
    int function(void*, in char*, int) _write;
    __sbuf _ub;
    ubyte* _up;
    int _ur;
    ubyte[3] _ubuf;
    ubyte[1] _nbuf;
    __sbuf _lb;
    int _blksize;
    fpos_t _offset;
    pthread_mutex_t _fl_mutex;
    pthread_t _fl_owner;
    int _fl_count;
    int _orientation;
    __mbstate_t _mbstate;
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}

}
    alias shared(_iobuf) FILE;
    enum 
{
_F_RDWR = 3,
_F_READ = 1,
_F_WRIT = 2,
_F_BUF = 4,
_F_LBUF = 8,
_F_ERR = 16,
_F_EOF = 32,
_F_BIN = 64,
_F_IN = 128,
_F_OUT = 256,
_F_TERM = 512,
}
    version (Windows)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 64,
_IONBF = 4,
_IOREAD = 1,
_IOWRT = 2,
_IOMYBUF = 8,
_IOEOF = 16,
_IOERR = 32,
_IOSTRG = 64,
_IORW = 128,
_IOTRAN = 256,
_IOAPP = 512,
}
    extern shared void function() _fcloseallp;

    private extern shared FILE[_NFILE] _iob;


    shared stdin = &_iob[0];
    shared stdout = &_iob[1];
    shared stderr = &_iob[2];
    shared stdaux = &_iob[3];
    shared stdprn = &_iob[4];
}
else
{
    version (linux)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 1,
_IONBF = 2,
}
    extern shared FILE* stdin;

    extern shared FILE* stdout;

    extern shared FILE* stderr;

}
else
{
    version (OSX)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 1,
_IONBF = 2,
}
    private extern shared FILE* __stdinp;


    private extern shared FILE* __stdoutp;


    private extern shared FILE* __stderrp;


    alias __stdinp stdin;
    alias __stdoutp stdout;
    alias __stderrp stderr;
}
else
{
    version (FreeBSD)
{
    enum 
{
_IOFBF = 0,
_IOLBF = 1,
_IONBF = 2,
}
    private extern shared FILE* __stdinp;


    private extern shared FILE* __stdoutp;


    private extern shared FILE* __stderrp;


    alias __stdinp stdin;
    alias __stdoutp stdout;
    alias __stderrp stderr;
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}
    alias int fpos_t;
    int remove(in char* filename);
    int rename(in char* from, in char* to);
    FILE* tmpfile();
    char* tmpnam(char* s);
    int fclose(FILE* stream);
    int fflush(FILE* stream);
    FILE* fopen(in char* filename, in char* mode);
    FILE* freopen(in char* filename, in char* mode, FILE* stream);
    void setbuf(FILE* stream, char* buf);
    int setvbuf(FILE* stream, char* buf, int mode, size_t size);
    int fprintf(FILE* stream, in char* format,...);
    int fscanf(FILE* stream, in char* format,...);
    int sprintf(char* s, in char* format,...);
    int sscanf(in char* s, in char* format,...);
    int vfprintf(FILE* stream, in char* format, va_list arg);
    int vfscanf(FILE* stream, in char* format, va_list arg);
    int vsprintf(char* s, in char* format, va_list arg);
    int vsscanf(in char* s, in char* format, va_list arg);
    int vprintf(in char* format, va_list arg);
    int vscanf(in char* format, va_list arg);
    int printf(in char* format,...);
    int scanf(in char* format,...);
    int fgetc(FILE* stream);
    int fputc(int c, FILE* stream);
    char* fgets(char* s, int n, FILE* stream);
    int fputs(in char* s, FILE* stream);
    char* gets(char* s);
    int puts(in char* s);
    extern (D) 
{
    int getchar()
{
return getc(stdin);
}
    int putchar(int c)
{
return putc(c,stdout);
}
    int getc(FILE* stream)
{
return fgetc(stream);
}
    int putc(int c, FILE* stream)
{
return fputc(c,stream);
}
}
    int ungetc(int c, FILE* stream);
    size_t fread(void* ptr, size_t size, size_t nmemb, FILE* stream);
    size_t fwrite(in void* ptr, size_t size, size_t nmemb, FILE* stream);
    int fgetpos(FILE* stream, fpos_t* pos);
    int fsetpos(FILE* stream, in fpos_t* pos);
    int fseek(FILE* stream, c_long offset, int whence);
    c_long ftell(FILE* stream);
    version (Windows)
{
    extern (D) 
{
    void rewind(FILE* stream)
{
fseek(stream,0L,SEEK_SET);
stream._flag &= ~_IOERR;
}
    void clearerr(FILE* stream)
{
stream._flag &= ~(_IOERR | _IOEOF);
}
    int feof(FILE* stream)
{
return stream._flag & _IOEOF;
}
    int ferror(FILE* stream)
{
return stream._flag & _IOERR;
}
}
    int _snprintf(char* s, size_t n, in char* fmt,...);
    alias _snprintf snprintf;
    int _vsnprintf(char* s, size_t n, in char* format, va_list arg);
    alias _vsnprintf vsnprintf;
}
else
{
    version (linux)
{
    void rewind(FILE* stream);
    void clearerr(FILE* stream);
    int feof(FILE* stream);
    int ferror(FILE* stream);
    int fileno(FILE*);
    int snprintf(char* s, size_t n, in char* format,...);
    int vsnprintf(char* s, size_t n, in char* format, va_list arg);
}
else
{
    version (OSX)
{
    void rewind(FILE*);
    void clearerr(FILE*);
    int feof(FILE*);
    int ferror(FILE*);
    int fileno(FILE*);
    int snprintf(char* s, size_t n, in char* format,...);
    int vsnprintf(char* s, size_t n, in char* format, va_list arg);
}
else
{
    version (FreeBSD)
{
    void rewind(FILE*);
    void clearerr(FILE*);
    int feof(FILE*);
    int ferror(FILE*);
    int fileno(FILE*);
    int snprintf(char* s, size_t n, in char* format,...);
    int vsnprintf(char* s, size_t n, in char* format, va_list arg);
}
else
{
    static assert(false,"Unsupported platform");
}
}
}
}
    void perror(in char* s);
}

