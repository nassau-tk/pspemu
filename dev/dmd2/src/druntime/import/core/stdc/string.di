// D import file generated from 'src\core\stdc\string.d'
module core.stdc.string;
private import core.stdc.stddef;

extern (C) nothrow @system 
{
    pure void* memchr(in void* s, int c, size_t n);

    pure int memcmp(in void* s1, in void* s2, size_t n);

    void* memcpy(void* s1, in void* s2, size_t n);
    void* memmove(void* s1, in void* s2, size_t n);
    void* memset(void* s, int c, size_t n);
    char* strcpy(char* s1, in char* s2);
    char* strncpy(char* s1, in char* s2, size_t n);
    char* strcat(char* s1, in char* s2);
    char* strncat(char* s1, in char* s2, size_t n);
    pure int strcmp(in char* s1, in char* s2);

    int strcoll(in char* s1, in char* s2);
    pure int strncmp(in char* s1, in char* s2, size_t n);

    size_t strxfrm(char* s1, in char* s2, size_t n);
    pure char* strchr(in char* s, int c);

    pure size_t strcspn(in char* s1, in char* s2);

    char* strpbrk(in char* s1, in char* s2);
    pure char* strrchr(in char* s, int c);

    pure size_t strspn(in char* s1, in char* s2);

    pure char* strstr(in char* s1, in char* s2);

    char* strtok(char* s1, in char* s2);
    char* strerror(int errnum);
    pure size_t strlen(in char* s);

}


