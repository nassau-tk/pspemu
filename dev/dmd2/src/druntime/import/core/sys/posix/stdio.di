// D import file generated from 'src\core\sys\posix\stdio.d'
module core.sys.posix.stdio;
private import core.sys.posix.config;

public import core.stdc.stdio;

public import core.sys.posix.sys.types;

extern (C) 
{
    version (linux)
{
    static if(__USE_LARGEFILE64)
{
    int fgetpos64(FILE*, fpos_t*);
    alias fgetpos64 fgetpos;
    FILE* fopen64(in char*, in char*);
    alias fopen64 fopen;
    FILE* freopen64(in char*, in char*, FILE*);
    alias freopen64 freopen;
    int fseek64(FILE*, c_long, int);
    alias fseek64 fseek;
    int fsetpos64(FILE*, in fpos_t*);
    alias fsetpos64 fsetpos;
    FILE* tmpfile64();
    alias tmpfile64 tmpfile;
}
else
{
    int fgetpos(FILE*, fpos_t*);
    FILE* fopen(in char*, in char*);
    FILE* freopen(in char*, in char*, FILE*);
    int fseek(FILE*, c_long, int);
    int fsetpos(FILE*, in fpos_t*);
    FILE* tmpfile();
}
}
    version (linux)
{
    enum L_ctermid = 9;
    static if(__USE_FILE_OFFSET64)
{
    int fseeko64(FILE*, off_t, int);
    alias fseeko64 fseeko;
}
else
{
    int fseeko(FILE*, off_t, int);
}
    static if(__USE_LARGEFILE64)
{
    off_t ftello64(FILE*);
    alias ftello64 ftello;
}
else
{
    off_t ftello(FILE*);
}
}
else
{
    version (Posix)
{
    int fseeko(FILE*, off_t, int);
    off_t ftello(FILE*);
}
}
    version (Posix)
{
    char* ctermid(char*);
    FILE* fdopen(int, in char*);
    int fileno(FILE*);
    char* gets(char*);
    int pclose(FILE*);
    FILE* popen(in char*, in char*);
}
    version (linux)
{
    void flockfile(FILE*);
    int ftrylockfile(FILE*);
    void funlockfile(FILE*);
    int getc_unlocked(FILE*);
    int getchar_unlocked();
    int putc_unlocked(int, FILE*);
    int putchar_unlocked(int);
}
    version (linux)
{
    enum P_tmpdir = "/tmp";
    char* tempnam(in char*, in char*);
}
}
