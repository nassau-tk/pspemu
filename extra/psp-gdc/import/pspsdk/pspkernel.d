module pspsdk.pspkernel;

/*
#include <pspuser.h>
#include <pspiofilemgr_kernel.h>
#include <psploadcore.h>
#include <pspstdio_kernel.h>
#include <pspsysreg.h>
#include <pspkdebug.h>
#include <pspintrman_kernel.h>
#include <pspmodulemgr_kernel.h>
*/

/**
 * Set the $pc register to a kernel memory address.
 *
 * When the PSP's kernel library stubs are called, they expect to be accessed
 * from the kernel's address space.  Use this function to set $pc to the kernel
 * address space, before calling a kernel library stub.
 */
/*
#define pspKernelSetKernelPC()  \
{     \
	__asm__ volatile (      \
	"la     $8, 1f\n\t"     \
	"lui    $9, 0x8000\n\t" \
	"or     $8, $9\n\t"     \
	"jr     $8\n\t"         \
	" nop\n\t"              \
	"1:\n\t"                \
	: : : "$8", "$9");      \
	sceKernelIcacheClearAll(); \
}
*/

/**
 * Set the $pc register to a user memory address.
 *
 */
/*
#define pspKernelSetUserPC()  \
{     \
	__asm__ volatile (      \
	"la     $8, 1f\n\t"     \
	"li     $9, 0x7FFFFFFF\n\t" \
	"and    $8, $9\n\t"     \
	"jr     $8\n\t"         \
	" nop\n\t"              \
	"1:\n\t"                \
	: : : "$8", "$9");      \
	sceKernelIcacheClearAll(); \
}
*/