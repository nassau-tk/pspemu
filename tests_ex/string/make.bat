@ECHO OFF
CALL ..\prepare.bat
psp-gcc -I. -I"%PSPSDK%/psp/sdk/include" -L. -L"%PSPSDK%/psp/sdk/lib" -D_PSP_FW_VERSION=150 -Wall -g -O0 string.c ../emits.c -lpspdebug -lpspge -lpspdisplay -lpspsdk -lc -lpspuser -lpspkernel -o string.elf
IF EXIST string.elf (
	psp-fixup-imports string.elf
)