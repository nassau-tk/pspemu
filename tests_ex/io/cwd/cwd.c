/*
 * PSP Software Development Kit - http://www.pspdev.org
 * -----------------------------------------------------------------------
 * Licensed under the BSD license, see LICENSE in PSPSDK root for details.
 *
 * main.c - Basic sample to demonstrate some fileio functionality.
 *
 * Copyright (c) 2005 Marcus R. Brown <mrbrown@ocgnet.org>
 * Copyright (c) 2005 James Forshaw <tyranid@gmail.com>
 * Copyright (c) 2005 John Kelley <ps2dev@kelley.ca>
 * Copyright (c) 2005 Jim Paris <jim@jtan.com>
 *
 * $Id: main.c 1175 2005-10-20 15:41:33Z chip $
 */
#include <pspkernel.h>
#include <pspctrl.h>
#include <pspdebug.h>
#include <pspdisplay.h>
#include <psptypes.h>
#include <pspiofilemgr.h>

#include <stdio.h>
#include <string.h>
#include <sys/types.h>
#include <sys/unistd.h>

#define eprintf(...) pspDebugScreenPrintf(__VA_ARGS__); Kprintf(__VA_ARGS__);

PSP_MODULE_INFO("CwdTest", 0, 1, 1);

void try(const char *dest)
{
	char buf[MAXPATHLEN];

	eprintf("%16s --> ", dest);
	if(chdir(dest) < 0) {
		eprintf("(chdir error)\n");
	} else {
		eprintf("%s\n", getcwd(buf, MAXPATHLEN) ?: "(getcwd error)");
	}
}

int main(int argc, char *argv[])
{
	int n;
	char buf[MAXPATHLEN];
	
	pspDebugScreenInit();

	eprintf("Working Directory Examples\n");
	eprintf("Arguments: %d\n", argc);
	for (n = 0; n < argc; n++) {
		eprintf("Argument[%d]: '%s'\n", n, argv[n]);
	}
	eprintf("Initial dir: %s\n\n", getcwd(buf, MAXPATHLEN) ?: "(error)");

	eprintf("%16s --> %s\n", "chdir() attempt", "resulting getcwd()");
	eprintf("%16s --> %s\n", "---------------", "------------------");
	try("");		   /* empty string                */
	try("hello");		   /* nonexistent path            */
	try("..");		   /* parent dir                  */
	try("../SAVEDATA");	   /* parent dir and subdir       */
	try("../..");		   /* multiple parents            */
	try(".");		   /* current dir                 */
	try("./././//PSP");        /* current dirs, extra slashes */
	try("/PSP/./GAME");	   /* absolute with no drive      */
	try("/");                  /* root with no drive          */
	try("ms0:/PSP/GAME");      /* absolute with drive         */
	try("flash0:/");           /* different drive             */
	try("ms0:/PSP/../PSP/");   /* mixed                       */

	eprintf("\nAll done!\n");

	return 0;
}
