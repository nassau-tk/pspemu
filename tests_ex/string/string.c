#include <pspkernel.h>
#include <pspthreadman.h>
#include <stdio.h>
#include <string.h>

PSP_MODULE_INFO("STRING TEST", 0, 1, 1);
PSP_MAIN_THREAD_ATTR(THREAD_ATTR_USER | THREAD_ATTR_VFPU);

float value = 13.0;

int main(int argc, char** argv) {
	char buffer[128];
	
	sprintf(buffer, "%d", (u32)0);
	Kprintf("%s\n", buffer);
	
	sprintf(buffer, "%d", (u32)100000);
	Kprintf("%s\n", buffer);

	sprintf(buffer, "%lld", (u64)9);
	Kprintf("%s\n", buffer);

	sprintf(buffer, "%lld", (u64)100000);
	Kprintf("%s\n", buffer);

	sprintf(buffer, "%.2f", value);
	Kprintf("%s\n", buffer);

	sceKernelExitGame();
	return 0;
}
