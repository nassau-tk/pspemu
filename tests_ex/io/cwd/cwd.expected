Working Directory Examples
Arguments: 1
Argument[0]: 'ms0:/PSP/GAME/virtual/cwd.elf'
Initial dir: ms0:/PSP/GAME/virtual

 chdir() attempt --> resulting getcwd()
 --------------- --> ------------------
                 --> ms0:/PSP/GAME/virtual
           hello --> (chdir error)
              .. --> ms0:/PSP/GAME
     ../SAVEDATA --> ms0:/PSP/SAVEDATA
           ../.. --> ms0:
               . --> ms0:
     ./././//PSP --> ms0:/PSP
     /PSP/./GAME --> ms0:/PSP/GAME
               / --> ms0:
   ms0:/PSP/GAME --> ms0:/PSP/GAME
        flash0:/ --> flash0:
ms0:/PSP/../PSP/ --> ms0:/PSP

All done!